### PR TITLE
view-mode toolbar

### PR DESCRIPTION
**Pull request by nelsonpecora**
* [ticket for toolbar](https://trello.com/c/wEar3qI1/18-view-edit-toolbar-toggle)
* [ticket for publish pane](https://trello.com/c/F00EswvT/34-view-mode-publish-pane)
* new toolbar! it displays when logged in
* new page and components pane open on the left side now
* includes stripped-down publish pane when in view mode (allows unpub/unsched, validation, toggling into edit mode. no publishing, scheduling, or custom location)

Notes:

* real history will replace the short publish state messages in the publish pane when history is ready
* status message updates are in a [separate ticket](https://trello.com/c/EYkRev1i/62-clay-status-messages)
* no animation with the inner toolbar, since it's a page reload
* removed "user" button and "history" button (they were hidden in the toolbar before)
* it uses separate controllers for the view/edit toolbar so we can have two different dependency trees
* pulled out toggling between view and edit modes into a service

screen shot 2016-11-02 at 1 17 57 pm

screen shot 2016-11-02 at 1 37 46 pm

screen shot 2016-11-02 at 2 26 21 pm

### CODE REVIEW

File: edit.js [hunk 1]
```diff
@@ -58,23 +57,19 @@ decorators.add(require('./decorators/component-list'));
 // note: external behaviors, decorators, and validation rules should already be added
 // when this event fires
 document.addEventListener('DOMContentLoaded', function () {
-  var parsed = nodeUrl.parse(location.href, true, true);
-
-  if (parsed.query.edit) {
```
Comment (nelsonpecora) [inline edit.js]: this file ONLY runs in edit mode, so we don't need this check anymore

File: services/page-state.js [hunk 2]
```diff
@@ -148,8 +149,8 @@ function getPageState() {
   var pageUri = dom.pageUri();
 
   return Promise.all([
-    getScheduled(pageUri + '@scheduled'),
-    getPublished(pageUri + '@published'),
+    getScheduled(references.replaceVersion(pageUri, 'scheduled')),
+    getPublished(references.replaceVersion(pageUri, 'published')),
```
Comment (nelsonpecora) [inline services/page-state.js]: this prevents double `@published@published` when running in view mode

File: media/close.svg [hunk 3]
```diff
@@ -1 +1 @@
-<svg width="22" height="23" viewBox="0 0 22 23" xmlns="http://www.w3.org/2000/svg"><path d="M12.667 11.24l9.388-9.388-1.61-1.61-9.388 9.388L1.67.242.06 1.852l9.388 9.388L.06 20.627l1.61 1.61 9.387-9.388 9.388 9.388 1.61-1.61-9.388-9.387z" fill="#C3C3C3" fill-rule="evenodd"/></svg>
+<svg width="22" height="23" viewBox="0 0 22 23" xmlns="http://www.w3.org/2000/svg"><path d="M12.667 11.24l9.388-9.388-1.61-1.61-9.388 9.388L1.67.242.06 1.852l9.388 9.388L.06 20.627l1.61 1.61 9.387-9.388 9.388 9.388 1.61-1.61-9.388-9.387z"/></svg>
```
Comment (gloddy) [inline media/close.svg]: Is this overwriting a different close button used elsewhere?

File: services/pane/publish-view.js [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+
+/**
+ * create actions for undoing scheduling and publishing
+ * @param {object} res
+ * @returns {Element}
+ */
+function createUndoActions(res) {
+  const undo = tpl.get('.publish-undo-template');
+
+  let unpublish, unschedule;
```
Comment (gloddy) [inline services/pane/publish-view.js]: Same line?

File: services/pane/publish-view.js [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+}
+
+/**
+ * create messages for the publish pane, depending on the state
+ * @param {object} res
+ * @returns {Element}
+ */
+function createPublishMessages(res) {
+  var messages = tpl.get('.publish-messages-template'),
+    scheduleMessage, stateMessage;
```
Comment (gloddy) [inline services/pane/publish-view.js]: Same line?

File: media/close.svg (hunk 3 above)
Comment (nelsonpecora) [inline media/close.svg]: Looks like that icon isn't used anywhere anymore. removing...

File: services/pane/publish-view.js (hunk 5 above)
Comment (nelsonpecora) [inline services/pane/publish-view.js]: same here

**Pull request merged**